### PR TITLE
Show correct clear/CopyDst/CopySrc subresource in texture viewer

### PR DESCRIPTION
**Pull request by RanXacT**
This is working on d3d12 for now. 
May as well get feedback before going too much further.

### CODE REVIEW

File: renderdoc/api/replay/data_types.h [hunk 1]
```diff
@@ -1360,10 +1404,15 @@ Valid values are 1 (depending on API), 2 or 4, or 0 if the drawcall is not an in
 operation.
 )");
   ResourceId copySource;
+  DOCUMENT(R"(The :class:`Subresource` specifying which part in copySource is used.)");
```
Comment (baldurk) [inline renderdoc/api/replay/data_types.h]: It would be nice to update the docstring to :data:\`copySource\` to link up to the other member.

**Review by baldurk — APPROVED** [review]
This looks good. Are you OK to implement for any of the other APIs?

File: renderdoc/api/replay/data_types.h (hunk 1 above)
Comment (RanXacT) [reply renderdoc/api/replay/data_types.h]: np.

**Comment by RanXacT** [comment]
> This looks good. Are you OK to implement for any of the other APIs?

I can do vk and test it. dx11 as well. As for gl, I can take a blind stab but I wouldn't be able to readily test.

**Comment by baldurk** [comment]
I can do the implementation for any API that you don't, it would be easy enough to make a synthetic test in the functional test suite.

**Comment by RanXacT** [comment]
kk. I'll leave gl to you then. I'll take. stab at the others in the next day or so.

File: renderdoc/api/replay/data_types.h (hunk 1 above)
Comment (RanXacT) [reply renderdoc/api/replay/data_types.h]: Should be good now.

**Review by RanXacT — COMMENTED** [review]
OK vk/d3d11/d3d12 are implemented and tested. 

Note that in testing I found an issue where `vkCmdBeginRenderPass(clear)` doesn't seem to show the cleared surface contents, it shows the prior contents of the surface.  Seems like this could be addressed in a separate commit though.

**Comment by RanXacT** [comment]
The travis-ci seems to fail `buildling osx` with:

> An error occurred while generating the build script.

But this seems like an infrastructure issue. I don't believe that error is due to the change.

File: renderdoc/driver/d3d11/d3d11_context1_wrap.cpp [hunk 2]
```diff
@@ -460,7 +460,15 @@ bool WrappedID3D11DeviceContext::Serialise_CopySubresourceRegion1(
       if(pDstResource && pSrcResource)
       {
         draw.copySource = srcOrigID;
+        draw.copySourceSubresource = Subresource(
+            GetMipForSubresource(GetResourceManager()->UnwrapResource(pSrcResource), SrcSubresource),
```
Comment (baldurk) [inline renderdoc/driver/d3d11/d3d11_context1_wrap.cpp]: There's no need to unwrap before calling `Get*ForSubresource` functions, which should make this a little more compact and easy to read.

**Review by baldurk — CHANGES_REQUESTED** [review]
I've made a minor comment about a tidy-up, but otherwise this looks good.

Looking into the vulkan `vkCmdBeginRenderPass` clear issue I've found the problem, so I've got a fix for that in the pipeline. Once this is merged I'll add handling for GL as well.

File: renderdoc/driver/d3d11/d3d11_context1_wrap.cpp (hunk 2 above)
Comment (RanXacT) [reply renderdoc/driver/d3d11/d3d11_context1_wrap.cpp]: sounds good. I'll get back on this later in the week (or over the weekend).

Comment (RanXacT) [reply renderdoc/driver/d3d11/d3d11_context1_wrap.cpp]: Done. re-tested and rebased on latest.

**Review by baldurk — APPROVED** [review]

**Pull request merged**